### PR TITLE
Switch to js-yaml for the YAML parser

### PR DESCRIPTION
**Pull request by jwforres**
Fixes https://github.com/openshift/origin/issues/10727

### CODE REVIEW

**Comment by jwforres** [comment]
@spadgett i did not add the loadAll support yet for multi-document YAML files in import YAML, I'd rather do that separately since it touches a lot more of the import YAML flow

**Comment by jwforres** [comment]
I'm just using error.message since it's already got everything i'd expect in it

yaml-error-display

**Comment by jwforres** [comment]
also as a note, i am using safeLoad and safeDump with the DEFAULT_SAFE_SCHEMA, more details: https://github.com/nodeca/js-yaml#safeload-string---options-

File: app/views/_parse-error.html [hunk 1]
```diff
@@ -4,9 +4,5 @@
   </button>
   <span class="pficon pficon-error-circle-o"></span>
   <strong>Failed to process the resource.</strong>
-  <div>
-    <truncate-long-text ng-if="error.message" content="error.message" limit="512" use-word-boundary="true"></truncate-long-text>
-    <span ng-if="error.parsedLine">Line: {{error.parsedLine}}</span>
-    <code ng-if="error.snippet">{{error.snippet}}</code>
-  </div>
+  <div style="white-space: pre;" ng-if="error.message">{{error.message}}</div>
```
Comment (spadgett) [inline app/views/_parse-error.html]: `class="pre-wrap"`?

**Comment by spadgett** [comment]
Should we use `safeLoadAll`? I believe that will handle multiple documents separated by `---`

https://github.com/nodeca/js-yaml#safeloadall-string-iterator---options-

**Comment by spadgett** [comment]
> Should we use safeLoadAll?

For import YAML I mean.

**Comment by spadgett** [comment]
Nevermind, just saw your comment above :)

LGTM, just one nit on the style.

**Comment by jwforres** [comment]
fixed the style
[merge]

**Pull request merged**